core/sts-3.3.md

```diff
@@ -0,0 +1,189 @@
+---
+title: IRCv3.3 Strict Transport Security
+layout: spec
+copyrights:
+  -
+    name: "Attila Molnar"
+    period: "2015"
+    email: "attilamolnar@hush.com"
+---
+
+## Description
+
+Strict Transport Security (STS) is a mechanism which allows servers to
+communicate that complying clients should only connect to them over a secure
+connection.
+
+The policy is communicated to the client via the `sts` capability and should
+be processed by the client at capability negotiation time.
+
+The value of the capability specifies the duration during which the client
+MUST only connect to the server securely (using TLS, aka. SSL), and the port
+number to use for it.
+
+Clients MUST use sufficient certificate verification to ensure the connection
+is secure without any errors.
+
+## Details
+
+The capability value, which MUST always be present, is a comma (`,`) separated
+list of tokens. Each token is a key which may have a value attached. If there
+is a value attached, the value is separated from the key by a `=`. That is,
+`<key>[=<value>][,<key2>[=<value2>][,<keyN>[=<valueN>]]]`. Keys specified in
+this document MUST only occur at most once.
+
+Clients MUST ignore every token with a key that they don't understand.
+
+An `sts` capability having at least a `duration` key expresses an STS policy.
+
+See [capability negotiation 3.2](capability-negotiation-3.2.html) for more
+information about capabilities with values.
+
+### Mechanism
+
+When the client sees the `sts` capability advertised over a non-secure
+connection, it MUST first establish a secure connection and confirm that the
+`sts` capability is still present.
```

Comment:
IRC:

```
2015-11-24 14:58:42+0200 < Not-6fa5> [ircv3-specifications] Mikaela reviewed pull request #155 commit - http://git.io/vB3ma
2015-11-24 15:03:20+0200 < Mikaela> Should that just say that the client should disconnect and not reconnect automatically (how does it fit the "don't ask the user"?) and port fixed in the example and STS is ready? Or are there more issues with it?
2015-11-24 16:54:57+0200 < Not-6fa5> [ircv3-specifications] jwheare reviewed pull request #155 commit - http://git.io/vBsIB
2015-11-24 17:06:23+0200 < Mikaela> The client knows that there is a MITM as the STS cap is not visible on the TLS connection
2015-11-24 17:06:46+0200 < jwheare> could just be a badly configured server and no mirm
2015-11-24 17:07:09+0200 < Mikaela> But in that case it's just adding port to example and it's ready for merge?
2015-11-24 17:07:25+0200 < Mikaela> AttilaM?
2015-11-24 17:07:27+0200 < jwheare> in these situations, clients should behave in simple predictable ways without trying to double guess, so that server admins can fix their problems properly
2015-11-24 17:07:55+0200 < Mikaela> Sounds good to me
2015-11-24 17:08:05+0200 < dx> if there's a MITM of the TLS connection, your root certificate store is compromised and you have no security at all
2015-11-24 17:08:22+0200 < dx> that's not a case we should be caring about
2015-11-24 17:09:21+0200 < Mikaela> So port is added to that example and it can be merged?
2015-11-24 19:43:11+0200 < Jobe> <Mikaela> The client knows that there is a MITM as the STS cap is not visible on the TLS connection
2015-11-24 19:43:25+0200 < Jobe> but whats to stop the MITM from listing STS in the cap response?
2015-11-24 19:43:52+0200 < Jobe> I mean if it has an unecrypted access to the C2S steam, it could manipulate what is sent anyway
2015-11-24 20:23:52+0200 < Mikaela> From what I understood that is ignored and it's just fixing the example before it gets ready for merging. But I would like confirmation now maybe three times.              
```